app/transport/rest/user_test.go

```diff
@@ -0,0 +1,112 @@
+package rest
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/go-devs-ua/octagon/app/entities"
+	"github.com/go-devs-ua/octagon/app/globals"
+	"github.com/go-devs-ua/octagon/lgr"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUserHandler_DeleteUser(t *testing.T) {
+	logger, err := lgr.New("INFO")
+	if err != nil {
+		t.FailNow()
+	}
+
+	tests := map[string]struct {
+		requestBody        string
+		usecaseConstructor func(ctrl *gomock.Controller) UserUsecase
+		expStatusCode      int
+		expResponseBody    string
+	}{
+		"success": {
+			requestBody: `{"id": "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc"}`,
+			usecaseConstructor: func(ctrl *gomock.Controller) UserUsecase {
+				mock := NewMockUserUsecase(ctrl)
+
+				mock.EXPECT().Delete(entities.User{
+					ID: "dca5947d-3dfc-49f1-bc09-dd53ce7e71cc",
+				}).Return(nil).Times(1)
+
+				return mock
+			},
+			expResponseBody: "",
+			expStatusCode:   http.StatusNoContent,
+		},
+		"invalid_undefined_id": {
```

Comment:
May be `notexisted-id` is better name for this test case?

Reply:
Done